hw3/hw3-react/src/App.js

```diff
@@ -0,0 +1,29 @@
+import './App.css';
+import {
+  BrowserRouter as Router,
+  Routes as Switch,
+  Route,
+  Redirect,
+} from "react-router-dom";
+import Home from './components/Home';
+import Search from './components/Search';
+import Houses from './components/Houses';
+import NavBar from './components/NavBar';
+
+function App() {
```

Comment:
It seems like you're making the API call multiple times in both the Search.js and HouseChart.jsx, which although works, is redundant and can take a hit on performance and efficiency. Since with React Browser Router, it mounts and unmounts the component that is rendered, your approach is making the API call every time a user clicks the Search navigation or House navigation. I'd recommend making the API call in the parent here (App.js) and pass the data as props to prevent this behavior.

Reply:
I agree with Hali on this one. You could fetch data in App.js and pass it to the other files as props.